lib/src/source/git.dart

```diff
@@ -253,9 +302,20 @@ class GitSource extends CachedSource {
   /// by [id] on the effective ref of [id].
   ///
   /// This assumes that the canonical clone already exists.
-  Future<String> _getRev(PackageId id) {
-    return git.run(["rev-list", "--max-count=1", _getEffectiveRef(id)],
-        workingDir: _repoCachePath(id)).then((result) => result.first);
+  Future<String> _getRev(PackageId id) async {
+    var ref = _getEffectiveRef(id);
+    return git
+        .run(["rev-list", "--max-count=1", ref], workingDir: _repoCachePath(id))
+        .then((result) => result.first)
```

Comment:
By async, I also meant getting red of `.then` and `.catchError` and using `await`, too. :smile: